packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -255,45 +258,41 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
                         headers: Object.keys(headers).length !== 0 ? true : undefined,
                         waitQueueLength: this.epochTracker.rateLimiter.waitQueueLength,
                     },
-                    async () => this.epochTracker.fetchAndParseAsJSON<IBlob>(url, { headers }, FetchType.blob),
+                    async () => this.epochTracker.fetchResponse(url, { headers }, FetchType.blob),
                 );
             });
-            blob = response.content;
+            buffer = await response.arrayBuffer();
         }
 
-        assert(blob.encoding === "base64" || blob.encoding === undefined, "wrong blob encoding format");
-        const inputFormat = blob.encoding;
-        if (!this.attributesBlobHandles.has(blobid)) {
-            if (outputFormat === inputFormat || (outputFormat === "string" && inputFormat === undefined))  {
-                return blob.content;
-            }
-            else if (outputFormat === "base64") {
-                return fromUtf8ToBase64(blob.content);
-            }
-            else {
-                return fromBase64ToUtf8(blob.content);
+        if (!this.attributesBlobHandles.has(blobId)) {
+            if (blob) {
+                if (outputFormat === blob.encoding || (outputFormat === "string" && blob.encoding === undefined))  {
+                    return blob.content;
+                } else if (outputFormat === "base64") {
+                    return fromUtf8ToBase64(blob.content);
+                } else {
+                    return fromBase64ToUtf8(blob.content);
+                }
             }
+
+            return IsoBuffer.from(buffer).toString(outputFormat === "base64" ? "base64" : "utf8");
```

Comment:
Conversion to UTF8 does not look right.

[RFC 4627 ](https://tools.ietf.org/html/rfc4627)(see Chapter 3 - Encoding) that governs application/json content type says that payload can be in any 5 formats (utf-8/16/32, BE or LE). When SPO receives such summary payload, I'd guess they will get proper in-memory representation (likely UTF-16, given that's default for most environments on Windows), and will likely store it in that format. Thus when we read data, it's likely to give us data in this format. They can move to Linux boxes in the future and accidently get UTF-32 (default everywhere else except Windows).

I guess the key - current protocol does not describe what encoding they store data in, and if that has material impact on this API. We can change it (and ensure it's part of protocol in some form or shape). If not, this code should auto detect encoding in the same way as RFC 4627 instructs browsers (and other entities) to treat Unicode formatting

Reply:
That all said, I think we are already in this world before your changes. Receiving "base64" input (as we get today) does not say how original buffer was generated, i.e. was it utf-8 or utf-16 before encoding it with base64. Our (runtime) utilities use utf-8, but when we send application/json payload, and when server processes input, it loses encoding (or rather does not provide us it). So I guess it's already utf-8 all the way through. I'll talk to SPO folks about it